### DISCUSSION TITLE
Stabilization Strategies

### DISCUSSION THREAD

**Discussion by 0x4007** [pavlovcik/uad-ui#218] [Ideas]
# Stabilization Strategies

## `$ 1.010+`
- uCR redemptions honored to mint new uAD supply - we should consider including a (e.g. 0.5%) redemption fee to profit for Ubiquity
- Auto arbitrage from whitelisted markets (pools economy wide liquidity) to frontrun and profit for Ubiquity
- If more liquidity in the metapool than all outstanding uCR cash redeem-ability amount, should refine buyback and LP mechanism on UBQ-uAD to "bank profits" basically funneling the value from the excess demand of the Dollar into propping up UBQ price.
## `$ 1.005`
- Withdraw from holdings of the "Ubiquity Bank" (cash sidelined as collateral, should be LPing somewhere earning yield) this is simply pulling from our collateral reserve and paying the seller from there. If we are 1:1 backed then the only time we _should_ go below peg is due to some miscalculation somewhere. Otherwise, assuming correct implementation/design; we can venture into under-collateralized world with the remaining strategies below.
## `$ 1.000-`
- Remove/burn uAD in market to maintain 1:1 parity on sells. Stableswap allows us to remove single-sided liquidity.
- Allow users to burn uAD for uCR (a weak lever that I imagine will make more sense when our economy is much larger and more mature.)
- Mint uCR as a debt because we depegged. Ideally we should give it to LPs if they withdraw below peg, but perhaps simpler to send to the seller. We owe them money because we broke our promise of 1:1 uAD -> USDC etc. We might be able to market this technology as "divergence loss protection" if we can keep track of the total entry and exit values of the LP position. Ideally if they can be minted the uCR upon sell of the withdrawn uAD into another stablecoin (this would realize their losses of depegged uAD) 

### Remarks

1. I think above peg we should take some profits too either directly via transfer hook or indirectly via credits e.g. redemption fee for uCR is a percentage like 0.5%
2. The complexity of transfer hooks could make a compelling case for l2/optimism
3. We should capture all the LP fees from the liquidity controlled by the staking contract, and the revenue from bonds, and set aside those as profits.
4. Perhaps we can design these bands to be flexible, where we can compress these policies to activate in a tighter price range as liquidity grows e.g. at `$1.0050`, `$1.0025`, `$1.0000`

**Comment by 0x4007** [comment]
Only can print new supply of dollars based on the following factors in order to be able to afford a 100% bank run: 

1. Collateral in the bank (independent of TWAP etc) 
2. Dollars circulating (outside of the sole control of Ubiquity DAO)

However we should be deeply looking into ways to raise collateral (profit) by 

1. Leveraging transfer hooks to front-run and arbitrage (and profit) 
  a. Arbs can only realistically occur between Curve and Uniswap pools (Uniswap and Curve routers handle arbs amongst their respective ecosystems)
2. LP swap fees of the liquidity that we control
3. Farming (e.g. Curve wars using our extra liquidity)
4. Revenue generating products

**Comment by 0x4007** [comment]
peg defense mechanics draft

user redeems uAD when we are below peg, there is a 0.5% fee(?) + user pays high gas fees for the following:
- dollarRedeemer checks for spare/idle stablecoin collateral to directly redeem (LUSD)
- if not enough, then withdraw yield from yield strategy (thinking uniswap v3 position via standard ERC4626 vault interface)
- if not enough, then withdraw principle from yield strategy
- if not enough, then dump UBQ for LUSD
- if not enough, then `priceReset` on 3pool

note: consider dydx rollup strategy where users can request withdraw on occasion or pay the fee for instant

main revenue strategies:
        1. yield strategy ERC4626 vault - e.g. ETH-uAD LP 0.3% (1000-2000 can be around 30% APR potentially)
                maybe only farm with the amount that we can afford to dump UBQ for in case of bankrun
        2. mint/redeem fee 0.5%
        3. uad-3crv swap fees at 0.05%
        4. curve bribes ponzi games

**Reply by 0x4007** [reply]
uCR should only redeem from yield/profit and not collateral principle. this means it should have claim over a portion of UBQ if all yield/profit will be funneled to buyback and LP UBQ.

**Reply by 0x4007** [reply]
"bonding curve of the redemption market" cool idea - https://docs.gyro.finance/gyroscope-protocol/overview

**Reply by 0x4007** [reply]
Auctioning uBOND off and letting the buyer set their yield/premium (e.g. 10% or 100%.) The catch could be that the protocol pays the cheaper uBONDs off first.

**Comment by sergfeldman** [comment]
For DevPool consider vaults Partner token - uAD LP
